utils/ChromiumDownloader.js

```diff
@@ -27,11 +27,15 @@ const getProxyForUrl = require('proxy-from-env').getProxyForUrl;
 
 const DOWNLOADS_FOLDER = path.join(__dirname, '..', '.local-chromium');
 
+const DOWNLOAD_HOST = process.env.PUPPETEER_DOWNLOAD_HOST
```

Comment:
It's not reliable to access env in ChromiumDownloader: it could be run in multiple of different processes and scripts:
- utils/check_availability.js
- install.js
- lib/Browser.js

For example, reading .npm_config_puppeteer_download_host doesn't make sense anywhere but in install.js.

Instead, host should be passed in to the `canDownloadRevision`/`downloadRevision` methods.

Reply:
So when I pass the host into these two methods I'd make the checks for them on the calling side. I don't see the spawning of new processes so I would have the exact same environment variables as the ChromiumDownloader has. Isn't it more maintainable to have the check in the central script? Or am I missing that it could be called from different spawned processes?

Since I'm defaulting the values, there is no harm in checking the config var for other scripts too. If it's not set, it would just be undefined and fall back to the hardcoded hostname. Or is there a potential error I don' see?

Reply:
> I don't see the spawning of new processes so I would have the exact same environment variables as the ChromiumDownloader has. Isn't it more maintainable to have the check in the central script? 

Reading env variables is required only in `install.js`. Maintenance-wise, I don't see much difference. 

>  Or am I missing that it could be called from different spawned processes?

ChromiumDownloader is `require`'d from different scripts, each of which could be run under different environments.

> Since I'm defaulting the values, there is no harm in checking the config var for other scripts too. If it's not set, it would just be undefined and fall back to the hardcoded hostname. Or is there a potential error I don' see?

No error here. I don't like reading env in ChromiumDownloader.js since it violates a [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege).

Reply:
I do see your points. I was too focused that I would have to be passed in by main callees that I overlooked it only being needed by install.js. I will extract this ENV var handling 👍